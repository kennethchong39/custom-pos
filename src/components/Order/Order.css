.order-item {
  padding: 15px 10px;
  border: none;
  border-radius: 5%;
  background-color: #00d3b5;
  font-size: 1.2rem;
  text-align: center;
  margin: 0px 5px 15px;
}

.order-item.complete {
  background-color: orange;
}

.order-item h3 {
  margin: 0px;
}

.order-item-button {
  width: 80%;
  padding: 0.5rem;
  background-color: white;
  border: none;
  text-transform: uppercase;
}

.order-item-flex {
  display: flex;
  justify-content: space-between;
  margin: 0px 20px;
  font-size: 1.3rem;
  font-weight: 600;
}

.order-item-flex-light {
  display: flex;
  justify-content: space-between;
  margin: 0px 30px;
  font-size: 1.3rem;
  border-bottom: 1px solid black;
}
