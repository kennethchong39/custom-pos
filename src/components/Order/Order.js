import React, { useState } from 'react';

import './Order.css';

export default function Order(props) {
  // to keep track of the order state if it is complete
  const [isComplete, setIsComplete] = useState(false);

  // to update the is complete state once the button is clicked
  const handleComplete = () => {
    setIsComplete(!isComplete);
  };

  return (
    <div className={isComplete ? 'order-item complete' : 'order-item'}>
      <div className="order-item-flex">
        <h3>Order Number</h3>
        <h3># {props.index + 1}</h3>
      </div>
      {props.item.order.map((item, index) => {
        return (
          <div className="order-item-flex-light" key={index}>
            <p>{item.name}</p>
            <p>x {item.quantity}</p>
            <p>$ {item.price}</p>
          </div>
        );
      })}
      <div className="order-item-flex">
        <p>Total</p>
        <p>$ {props.item.total}</p>
      </div>
      <button
        disabled={isComplete}
        className={
          isComplete ? 'order-item-button disabled' : 'order-item-button'
        }
        onClick={handleComplete}
      >
        {isComplete ? 'Order Picked-Up' : 'Complete ?'}
      </button>
    </div>
  );
}
