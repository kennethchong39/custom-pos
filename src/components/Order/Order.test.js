import renderer from 'react-test-renderer';

import Order from './Order';

test('component: Order tests', () => {
  const testProps = {
    order: [
      {
        name: 'Vegetarian',
        price: 8.99,
        ingredients: {
          bread: 2,
          lettuce: 2,
          tomato: 2,
          cheese: 2,
        },
        quantity: 1,
      },
    ],
    total: 8.99,
  };
  const tree = renderer.create(<Order item={testProps} />).toJSON();

  expect(tree).toMatchSnapshot();
});
