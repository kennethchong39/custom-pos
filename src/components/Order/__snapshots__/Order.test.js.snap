// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: Order tests 1`] = `
<div
  className="order-item"
>
  <div
    className="order-item-flex"
  >
    <h3>
      Order Number
    </h3>
    <h3>
      # 
      NaN
    </h3>
  </div>
  <div
    className="order-item-flex-light"
  >
    <p>
      Vegetarian
    </p>
    <p>
      x 
      1
    </p>
    <p>
      $ 
      8.99
    </p>
  </div>
  <div
    className="order-item-flex"
  >
    <p>
      Total
    </p>
    <p>
      $ 
      8.99
    </p>
  </div>
  <button
    className="order-item-button"
    disabled={false}
    onClick={[Function]}
  >
    Complete ?
  </button>
</div>
`;
